File: jklulogger.py
from requests import post as post_request
from requests.structures import CaseInsensitiveDict
from requests.exceptions import ConnectTimeout

username = "YOUR_USERNAME"
password = "YOUR_PASSWORD"

url = "http://172.16.1.1:8090/login.xml"

headers = CaseInsensitiveDict()
headers["Accept"] = "*/*"
headers["Accept-Language"] = "en-GB,en-US;q=0.9,en;q=0.8"
headers["Connection"] = "keep-alive"
headers["Content-Type"] = "application/x-www-form-urlencoded"
headers["DNT"] = "1"
headers["Sec-GPC"] = "1"
headers["User-Agent"] = "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36"

data = f"mode=191&username={username}&password={password}&a=1653625623569&producttype=0"

try:
    post_request(url, headers=headers, data=data, timeout=5)
except ConnectTimeout:
    pass

File: jklulogger.java
import java.io.IOException;
import java.io.OutputStream;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.charset.StandardCharsets;

public class jklulogger {

    private static String username = "YOUR_USERNAME";
    private static String password = "YOUR_PASSWORD";

    public static void connect() throws IOException {
        URL url = new URL("http://172.16.1.1:8090/login.xml");
        HttpURLConnection http = (HttpURLConnection) url.openConnection();

        http.setRequestMethod("POST");
        http.setConnectTimeout(5000);
        http.setDoOutput(true);
        http.setRequestProperty("Accept", "*/*");
        http.setRequestProperty("Accept-Language", "en-GB,en-US;q=0.9,en;q=0.8");
        http.setRequestProperty("Connection", "keep-alive");
        http.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
        http.setRequestProperty("DNT", "1");
        http.setRequestProperty("Sec-GPC", "1");
        http.setRequestProperty("User-Agent",
                "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36");

        String data = String.format("mode=191&username=%s&password=%s&a=1653625623569&producttype=0", username,
                password);

        byte[] out = data.getBytes(StandardCharsets.UTF_8);
        OutputStream stream = http.getOutputStream();
        stream.write(out);

        System.out.println(http.getResponseCode() + " " + http.getResponseMessage());
        http.disconnect();
    }

    public static void main(String[] args) {
        try {
            connect();
        } catch (IOException e) {

        }
    }
}
